GPR183.py

```diff
@@ -0,0 +1,184 @@
+        if not (gc_check or len_check or quality_check):
+            filtered_fastq.append(line)
+
+    # Write  filtered data into new .fastq file
+    SeqIO.write(filtered_fastq, output_path, 'fastq')
+
+# Custom error
+class WrongSequence(ValueError):
+    pass
+
+
+class BiologicalSequence(str):
+    def __init__(self, sequence):
+        self.sequence = sequence
+
+    def __len__(self):
+        return len(self.sequence)
+    
+    def slice(self, start_index, stop_index):
+        return self.sequence[start_index:stop_index]
+    
+    def alphabet_checking(self):
+        if not set(self.sequence) <= set(type(self).dictionary.keys()):
+            raise WrongSequence('Wrong sequence')
+        return True
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    
+    def __init__(self, sequence):
+        super().__init__(sequence)
+        if not self.alphabet_checking():
+            del self.sequence
+            raise WrongSequence('You have entered a wrong sequence')
+        self.gc_cont = None
+
+    def complement(self):
+        return type(self)(''.join([type(self).dictionary[i] for i in self.sequence]))
+
+    def gc_content(self):
+        gc = 0
+        for nucleotide in self.sequence:
+            if nucleotide in set('CGcg'):
+                gc += 1
+        self.gc_cont = round(gc / len(self.sequence), 3)
+        return self.gc_cont
+
+
+class DNAsequence(NucleicAcidSequence):
+    dictionary = {
+        'A': 'T',
+        'G': 'C',
+        'T': 'A',
+        'C': 'G',
+        'a': 't',
+        'g': 'c',
+        't': 'a',
+        'c': 'g',
+}
+    trans_dict = {
+        'A': 'U',
+        'G': 'C',
+        'T': 'A',
+        'C': 'G',
+        'a': 'u',
+        'g': 'c',
+        't': 'a',
+        'c': 'g',
+}
+    def transcribe(self):
+        return RNAsequence(''.join([self.trans_dict[i] for i in self.sequence]))
+
+
+class RNAsequence(NucleicAcidSequence):
+    dictionary = {
+        'A': 'U',
+        'G': 'C',
+        'U': 'A',
+        'C': 'G',
+        'a': 'u',
+        'g': 'c',
+        'u': 'a',
+        'c': 'g',
+}
+
+
+class AminoAcidSequence(BiologicalSequence):
+    dictionary = {
+        "A": 71.03711,
+        "C": 103.00919,
+        "D": 115.02694,
+        "E": 129.04259,
+        "F": 147.06841,
+        "G": 57.02146,
+        "H": 137.05891,
+        "I": 113.08406,
+        "K": 128.09496,
+        "L": 113.08406,
+        "M": 131.04049,
+        "N": 114.04293,
+        "P": 97.05276,
+        "Q": 128.05858,
+        "R": 156.10111,
+        "S": 87.03203,
+        "T": 101.04768,
+        "V": 99.06841,
+        "W": 186.07931,
+        "Y": 163.06333,
+    }
+    def __init__(self, sequence):
+        super().__init__(sequence)
+        if not self.alphabet_checking():
+            del self.sequence
+            raise WrongSequence('You have entered a wrong sequence')
```

Comment:
Перед init должна быть одна строчка

```suggestion
 
       def __init__(self, sequence):
        super().__init__(sequence)
        if not self.alphabet_checking():
            raise WrongSequence('You have entered a wrong sequence')
```